File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;

    printf("O hai!");

    do
    {
        printf("How much change is owed?\n");
        change = get_float();
    } while (change < 0);

    int change_in_cents = round(change * 100);
    int change_given = 0;

    if (change_in_cents % 25 >= 0 )
    {
        change_given += floor(change_in_cents / 25);
        change_in_cents = change_in_cents % 25;
    }
    if (change_in_cents % 10 >= 0)
    {
        change_given += floor(change_in_cents / 10);
        change_in_cents = change_in_cents % 10;
    }
    if (change_in_cents % 5 >= 0)
    {
        change_given += floor(change_in_cents / 5);
        change_in_cents = change_in_cents % 5;
    }
    if (change_in_cents % 1 >= 0)
    {
        change_given += floor(change_in_cents / 1);
        change_in_cents = change_in_cents % 1;
    }

    printf("%i\n", change_given);
}

File: pset6/greedy.py
import cs50

def main():
    print("O hai!")
    # input how much change is owned
    while(True):
        print("How much change is owed?")
        change_in_cents = round(cs50.get_float() * 100)
        if (change_in_cents > 0):
            break

    print("Total change: {}".format(change_in_cents))
    # convert dollars into cents
    # declare a variable to count coins
    # using modulus and divisions calculate required coins starting from the largest denomination

if __name__ == "__main__":
    main()
